### PR TITLE
[Common] refactor info list

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Refactor functions for info-list, use linked list and add new function to append info.

### CODE REVIEW

File: c/src/ml-api-service-agent-client.c [hunk 1]
```diff
@@ -176,7 +183,7 @@ _build_ml_info_from_json_cstr (const gchar * jcstring, void **handle)
   } else {
     if (_info_list)
       ml_information_list_destroy (_info_list);
-    if (_info)
+    else if (_info)
       ml_information_destroy (_info);
```
Comment (songgot) [inline c/src/ml-api-service-agent-client.c:187]: in case of line 174, ```_info``` and ```_info_list``` need to free, right?

**Review by anyj0527 — APPROVED** [review]
LGTM👍

File: c/src/ml-api-service-agent-client.c (hunk 1 above)
Comment (jaeyun-jung) [reply c/src/ml-api-service-agent-client.c:187]: In the changes, _info is added into _info_list. So If given json is array, destroying _info_list internally releases _info.

Comment (songgot) [reply c/src/ml-api-service-agent-client.c:187]: Thank very much :pray:

**Pull request merged**